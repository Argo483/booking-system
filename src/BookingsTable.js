import React from "react";
import { BookingsTableRow } from "./BookingsTableRow";

export const BookingsTable = ({ calendarDays }) => {
  return (
    <div style={{ display: "flex", flexDirection: "column" }}>
      {calendarDays.map((calendarDay, index) => {
        return (
          <div
            key={index}
            style={{
              width: "1000px",
              margin: "5px",
              padding: "10px",
              backgroundColor: "mediumgrey",
            }}
          >
            {calendarDay.day.format("L")}
            <table
              style={{
                display: "flex",
                flexDirection: "column",
                backgroundColor: "grey",
              }}
            >
              <tbody>
                {calendarDay.hours.map((hourToRender, index) => {
                  return (
                    <BookingsTableRow
                      key={index}
                      hourToRender={hourToRender}
                      calendarDay={calendarDay}
                      index={index}
                    ></BookingsTableRow>
                  );
                })}
              </tbody>
            </table>
          </div>
        );
      })}
    </div>
  );
};
