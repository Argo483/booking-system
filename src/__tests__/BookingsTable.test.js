import React from "react";
import renderer from "react-test-renderer";
import moment from "moment";
import { BookingsTable } from "../BookingsTable";
import "@testing-library/jest-dom/extend-expect";

it("should render multiple calendar days", () => {
  const tree = renderer
    .create(
      <BookingsTable
        calendarDays={[getMockCalendarDay(), getMockCalendarDay()]}
      ></BookingsTable>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

const getMockCalendarDay = () => {
  const hourToRender = moment("20111031 01:00", "YYYYMMDD hh:mm");
  return {
    hourToRender,
    day: moment("20111031 00:00", "YYYYMMDD hh:mm"),
    bookingsOnThisDay: [
      {
        startTime: moment("20111031 01:30", "YYYYMMDD hh:mm"),
        endTime: moment("20111031 02:30", "YYYYMMDD hh:mm"),
      },
    ],
    csvBookingsOnThisDay: [
      {
        startTime: moment("20111031 01:30", "YYYYMMDD hh:mm"),
        endTime: moment("20111031 02:30", "YYYYMMDD hh:mm"),
      },
    ],
    hours: [hourToRender],
  };
};
