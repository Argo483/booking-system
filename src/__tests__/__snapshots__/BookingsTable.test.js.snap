// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render multiple calendar days 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "column",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "mediumgrey",
        "margin": "5px",
        "padding": "10px",
        "width": "1000px",
      }
    }
  >
    10/31/2011
    <table
      style={
        Object {
          "backgroundColor": "grey",
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <tbody>
        <tr
          style={
            Object {
              "backgroundColor": "lightgrey",
            }
          }
        >
          <td>
            1:00 AM
          </td>
          <td
            data-testid="current-booking"
            style={
              Object {
                "backgroundColor": "initial",
                "width": "100px",
              }
            }
          />
          <td
            style={
              Object {
                "backgroundColor": "initial",
                "width": "100px",
              }
            }
          />
        </tr>
      </tbody>
    </table>
  </div>
  <div
    style={
      Object {
        "backgroundColor": "mediumgrey",
        "margin": "5px",
        "padding": "10px",
        "width": "1000px",
      }
    }
  >
    10/31/2011
    <table
      style={
        Object {
          "backgroundColor": "grey",
          "display": "flex",
          "flexDirection": "column",
        }
      }
    >
      <tbody>
        <tr
          style={
            Object {
              "backgroundColor": "lightgrey",
            }
          }
        >
          <td>
            1:00 AM
          </td>
          <td
            data-testid="current-booking"
            style={
              Object {
                "backgroundColor": "initial",
                "width": "100px",
              }
            }
          />
          <td
            style={
              Object {
                "backgroundColor": "initial",
                "width": "100px",
              }
            }
          />
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
